feather/protocol/src/packets/server/play.rs

```diff
@@ -1179,6 +1181,163 @@ packets! {
+            EntityAttributeKind::ZombieSpawnReinforcements => "zombie:spawn_reinforcements",
+        })
+    }
+
+    pub fn max_value(&self) -> f32 {
+        match &self {
+            EntityAttributeKind::MaxHealth => 1024.0,
+            EntityAttributeKind::FollowRange => 2048.0,
+            EntityAttributeKind::KnockbackResistance => 1.0,
+            EntityAttributeKind::MovementSpeed => 1024.0,
+            EntityAttributeKind::AttackDamage => 2048.0,
+            EntityAttributeKind::AttackSpeed => 1024.0,
+            EntityAttributeKind::AttackKnockback => 5.0,
+            EntityAttributeKind::FlyingSpeed => 1024.0,
+            EntityAttributeKind::Armor => 30.0,
+            EntityAttributeKind::ArmorToughness => 20.0,
+            EntityAttributeKind::Luck => 1024.0,
+            EntityAttributeKind::HorseJumpStrength => 2.0,
+            EntityAttributeKind::ZombieSpawnReinforcements => 1.0,
+        }
+    }
+
+    pub fn min_value(&self) -> f32 {
+        if let EntityAttributeKind::Luck = &self {
+            return -1024.0;
+        }
+
+        0.0
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct AttributeModifier {
+    pub uuid: Uuid,
+    pub amount: f32,
+    pub operation: OperationKind,
+}
+
+#[derive(Debug, Clone)]
+pub struct EntityProperty {
+    pub attribute: EntityAttributeKind,
+    pub modifiers: Vec<AttributeModifier>,
+    pub value: f32,
+}
+
+impl Readable for EntityProperty {
+    fn read(
+        buffer: &mut std::io::Cursor<&[u8]>,
+        version: crate::ProtocolVersion,
+    ) -> anyhow::Result<Self>
+    where
+        Self: Sized,
+    {
+        let attribute_key = String::read(buffer, version)?;
+        let attribute = EntityAttributeKind::from_key(&attribute_key)?;
+        let value = f32::read(buffer, version)?;
+
+        let num_modifiers = VarInt::read(buffer, version)?.0;
+        let mut modifiers = Vec::new();
+
+        for _ in 0..num_modifiers {
+            let uuid = Uuid::read(buffer, version)?;
+            let amount = f32::read(buffer, version)?;
+            let operation = OperationKind::read(buffer, version)?;
+
+            modifiers.push(AttributeModifier {
+                uuid,
+                amount,
+                operation,
+            });
+        }
+
+        Ok(Self {
+            attribute,
+            modifiers,
+            value,
+        })
+    }
+}
+
+impl Writeable for EntityProperty {
+    fn write(&self, buffer: &mut Vec<u8>, version: crate::ProtocolVersion) {
+        self.attribute.id().write(buffer, version);
+        self.value.write(buffer, version);
+
+        let num_modifiers = self.modifiers.len() as i32;
+        VarInt(num_modifiers).write(buffer, version);
+
+        for modifier in &self.modifiers {
+            modifier.uuid.write(buffer, version);
+            modifier.amount.write(buffer, version);
+            modifier.operation.write(buffer, version);
+        }
+
+        println!("{:?}", buffer);
+    }
+}
+
```

Comment:
I didn't check these implementations as they should be unnecessary once you define `AttributeModifier`, `EntityProperty` and `EntityAttributeKind` with macros (See my other comments).